### PR TITLE
Feature/better alertmanager interface

### PR DESCRIPTION
**Pull request by dstathis**
Reopening Leon's pull request from my branch so I can make changes as necessary. Also added one commit to make it work right.

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -133,33 +134,8 @@ def _on_grafana_changed(self, event):
...
-
-        When an Alertmanager departs it is removed from the list
-        of currently available Alertmanagers, the Prometheus configuration
-        is updated and Prometheus is restarted.
-        """
-        if not self.unit.is_leader():
-            return
-        self._stored.alertmanagers.clear()
+    def _on_alertmanager_cluster_changed(self, event):
+        self._stored.alertmanagers = self.alertmanager_lib.get_cluster_info()
```
Comment (balbirthomas) [inline src/charm.py:138]: Now that you have `get_cluster_info()` it is no longer necessary to store this in two places that is in the Prometheus charm and the `AlertmanagerConsumer`. It should be possible to just invoke `get_cluster_info()` from within `_alerting_config()` method of the Prometheus charm.

**Comment by sed-i** [comment]
This change works well together with https://github.com/canonical/alertmanager-operator/pull/13

**Comment by rbarry82** [comment]
@sed-i Development paradigms have changed a lot in the last 3 weeks. Are we sure we want the consumer emitting events instead of updating the databag and relying on `relation_changed`?

**Comment by sed-i** [comment]
> @sed-i Development paradigms have changed a lot in the last 3 weeks. Are we sure we want the consumer emitting events instead of updating the databag and relying on `relation_changed`?

The use of custom events and StoredState is the most simple and concise solution I found to the challenges encountered. 
Relying on `relation_changed` being handled by the consumer charm means that implementation detail of alertmanager-operator side would be implicitly coded in prometheus. I would rather abstract it away with interfaces using the alertmanager library.

Can we take a deeper look into this via issues against alertmanager?

**Comment by sed-i** [comment]
moving back to https://github.com/canonical/prometheus-operator/pull/48 and closing this one

**Pull request closed without merging**